lib/hanami/cli/registry.rb

```diff
@@ -86,6 +86,24 @@ def get(arguments)
         @commands.get(arguments)
       end
 
+      # @since x.x.x
+      # @api private
+      def before(command, &callback)
+        command_class(command).before(&callback)
+      end
+
+      # @since x.x.x
+      # @api private
+      def after(command, &callback)
+        command_class(command).after(&callback)
```

Comment:
What I suggest to do is to register callbacks to `CommandRegistry`, which is referenced as `@commands` from this file.

Then both `CommandRegistry#get` and `#set` should manage callbacks.

Reply:
nice catch 💯